daemon/pod.go

```diff
@@ -728,35 +729,66 @@ func (p *Pod) setupMountsAndFiles(sd Storage) (err error) {
 
 func (p *Pod) mountVolumes(daemon *Daemon, sd Storage) (err error) {
 	err = nil
-	p.volumes = []*hypervisor.VolumeInfo{}
 
 	var (
 		sharedDir = path.Join(hypervisor.BaseDir, p.vm.Id, hypervisor.ShareDirTag)
 	)
 
 	for _, v := range p.spec.Volumes {
-		var vol *hypervisor.VolumeInfo
+		var volInfo *hypervisor.VolumeInfo
 		if v.Source == "" {
 			err = fmt.Errorf("volume %s in pod %s is not created", v.Name, p.id)
 			return err
 		}
 
-		vol, err = ProbeExistingVolume(&v, sharedDir)
+		volInfo, err = ProbeExistingVolume(&v, sharedDir)
 		if err != nil {
 			return err
 		}
 
-		p.volumes = append(p.volumes, vol)
+		if vol, ok := p.volumes[v.Name]; ok {
+			vol.Filepath = volInfo.Filepath
+			vol.Fstype = volInfo.Fstype
+			vol.Format = volInfo.Format
+			continue
+		}
+
+		p.volumes[v.Name] = volInfo
+	}
```

Comment:
I think only `p.volumes[v.Name] = volInfo` is Ok,  we don't need to copy it one field by field if the volume existed.